### DISCUSSION TITLE
Quake 2 Successful Build

### DISCUSSION THREAD

**Discussion by ferrellsl** [Joshua-Riek/ubuntu-rockchip#55] [General]
I initially tried installing Quake 2 from the snap store.  It works, but there's no audio so I thought I'd just build a new version from source and it was a fairly easy build.  I used the sources from this repo:  https://github.com/yquake2/yquake2

Performance is fantastic but stick to using the OpenGL 1.4 driver as the EGL driver expects a system using EGL3.2.  The EGL driver WILL work, but you will notice graphics glitches.  I haven't tried the software render but there's also an OpenGL 3.2 driver that won't work properly since Panfrost is currently only up to OpenGL 3.0.

**Comment by ferrellsl** [comment]
I don't recommend using this build of Quake 2 for any benchmarking.  The timedemo benchmark reports some odd fps numbers like 45 fps even though you can see that it's moving at over 400 fps....crazy fast.

**Comment by Fredobedo** [comment]
Following our discussion from https://github.com/Joshua-Riek/ubuntu-orange-pi5/discussions/71#discussioncomment-5326872
I successfully compiled it and can now play with the game.

2 little additions:
- sudo apt libcurl-dev didn't work for me, had to install libcurl4-openssl-dev instead.
- downloaded original game from GOG on my PC and copied not only baseq2 but also the sound tracks ('music' folder)

Many thanks!

**Reply by ferrellsl** [reply]
Glad to hear it.
